### ISSUE TITLE
Feature: dry mode that lists packages that can be upgraded

### ISSUE THREAD

**Issue by Merinorus** [alanhamlett/pip-update-requirements#50]
Hello,

First of all, many thanks for this tool which is easy and convenient to use!

Is there a possibility to have a kind-of dry mode that lists only packages to be upgraded?

For instance:
```shell
$> pur -r requirements.txt -d
fastapi: 0.80.0 -> 0.82.0
python-dotenv: 0.20.0 -> 0.21.0
2 package upgrade(s) available.
```

I tried something like:
```shell
$> pur -n -r requirements.txt -d | sed '$ d' | diff --suppress-common-lines -y requirements.txt -
                                                              > ==> requirements.txt <==
fastapi==0.80.0                                               | fastapi==0.82.0
python-dotenv==0.20.0                                         | python-dotenv==0.21.0
```

Or maybe there is another tool that already provides this information.

Thank you!

**Comment by alanhamlett** [comment]
Yes, this already exists with `-d`.

**Comment by Merinorus** [comment]
Hello,
Thanks for your feedback. But I meant a command that lists packages that need to be upgraded and **only** them. **Not all** the packages of the requirements file.

**Comment by alanhamlett** [comment]
Implemented in v7.1.0 with a new `--dry-run-changed` option.
